packages/next/src/server/config.ts

```diff
@@ -1382,7 +1411,58 @@ export default async function loadConfig(
       userConfig.htmlLimitedBots = userConfig.htmlLimitedBots.source
     }
 
-    onLoadUserConfig?.(Object.freeze(loadedConfig))
+    if (
+      debugPrerender &&
+      (phase === PHASE_PRODUCTION_BUILD || phase === PHASE_EXPORT)
+    ) {
+      userConfig.experimental ??= {}
```

Comment:
We can handle the undefined case in `setExperimentalFeatureForDebugPrerender` rather than mutate it outside. This could lead to confusion on why `userConfig.experimental` always being defined if we use it to check sth else later

Reply:
I prefer to assign it outside.

Reply:
We already experienced a bug of mutated next config here causing other issue, I prefer to keep it safe. You can keep it outside with fallback to a new variable but should avoid mutating it.

Reply:
My point is to avoid mutation in general, not because we cloned above so then it's fine to mutate everywhere. I dont know when I need to handle the new empty object `.experimental` property then found it's mutated here

Reply:
I don't understand, we're mutating the _cloned_ config all over the place in this file (assigning defaults, transferring configs from one place to another, etc.). I thought you wanted to only prevent mutating the _original_ config in #80573?

Reply:
We do need to mutate here though, that's the whole point. This is not for logging purposes, but we're actually setting those flags so that they are used when prerendering.